tests/Unleash.Tests/Strategy/RemoteAddressStrategyTest.cs

```diff
@@ -62,5 +62,65 @@ public void test(string actualIp, string parameterstring, bool expected)
             parameters.Add(RemoteAddressStrategy.PARAM, ipstring);
             return parameters;
         }
+
+        [Test]
+        public void FindsInCIDRRange()
+        {
+            var range = "73.125.227.0/29";
+            var input = "73.125.227.7";
+            var context = UnleashContext.New().RemoteAddress(input).Build();
+            var parameters = setupParameterMap(range);
+            strategy.IsEnabled(parameters, context).Should().BeTrue();
+        }
+
+        [Test]
+        public void MatchOnRangeAndSingleIP()
+        {
+            var range = "73.125.227.0/29,73.125.227.7";
+            var input = "73.125.227.7";
+            var context = UnleashContext.New().RemoteAddress(input).Build();
+            var parameters = setupParameterMap(range);
+            strategy.IsEnabled(parameters, context).Should().BeTrue();
+        }
+
+        [Test]
+        public void MatchOnSingleIPOutsideRange()
+        {
+            var range = "73.125.227.0/29,73.125.227.114";
+            var input = "73.125.227.114";
+            var context = UnleashContext.New().RemoteAddress(input).Build();
+            var parameters = setupParameterMap(range);
+            strategy.IsEnabled(parameters, context).Should().BeTrue();
+        }
+
+        [Test]
+        public void MatchOnRangeNotSingleIP()
+        {
+            var range = "73.125.227.0/29,73.125.227.114";
+            var input = "73.125.227.7";
+            var context = UnleashContext.New().RemoteAddress(input).Build();
+            var parameters = setupParameterMap(range);
+            strategy.IsEnabled(parameters, context).Should().BeTrue();
+        }
+
+        [Test]
+        public void DoesntMatchOnCIDRJunk()
+        {
+            var range = "73.125.227.0/junk";
+            var input = "73.125.227.7";
+            var context = UnleashContext.New().RemoteAddress(input).Build();
+            var parameters = setupParameterMap(range);
+            strategy.IsEnabled(parameters, context).Should().BeFalse();
+        }
+
+        [Test]
+        public void GuardsAgainstTooHighCIDR()
+        {
+            var range = "73.125.227.1/59";
+            var input = "73.125.227.1";
+            var context = UnleashContext.New().RemoteAddress(input).Build();
+            var parameters = setupParameterMap(range);
+            strategy.IsEnabled(parameters, context).Should().BeTrue();
+        }
     }
```

Comment:
The tests look good to me! I suggest adding another one check that the same strategy can be invoked multiple times and returns the same result, just to be defensive about the mutations we need in `IPCIDRAddressRange` so we don't get burned later

Reply:
Good thinking! adding!